jobbergate-api/jobbergate_api/apps/job_scripts/routers.py

```diff
@@ -287,21 +288,32 @@ async def job_script_update(job_script_id: int, job_script: JobScriptUpdateReque
     update_query = (
         job_scripts_table.update()
         .where(job_scripts_table.c.id == job_script_id)
-        .values(job_script.dict(exclude_unset=True))
+        .values(
+            job_script.dict(exclude_unset=True, exclude={"job_script_data_as_string"}),
+        )
         .returning(job_scripts_table)
     )
     try:
         result = await database.fetch_one(update_query)
     except INTEGRITY_CHECK_EXCEPTIONS as e:
         raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail=str(e))
+    else:
+        if job_script.job_script_data_as_string:
+            s3man_jobscripts.put(
```

Comment:
Should we update the database if any error occurs when uploading the job script to S3?

Reply:
@matheushent @dusktreader

Reply:
Since the script itself is the core of the entries  in "job_scripts", I think it makes sense to actually roll-back the database changes if the script could not be uploaded. I can't think of a case where it's very useful to have an entry in job_scripts but the script itself is not there.

I think we should do the same for the `create` logic as well. In a try/except block, create the row and upload the data. If any of the operations fails, rollback the database and throw an exception that provides the context of what went wrong.

Reply:
Absolutely.
Mental note: Write tests to verify the roll-back.

Reply:
I think this operation in the time consuming one. That said, it should be done first than the SQL query, this way we wouldn't spend unnecessary SQL queries in case there's error uploading to S3

Reply:
Well, I also see the opposite. If we upload first to S3 and something goes wrong with the database, we will have problems rolling back the file at S3 (it is not possible, as far as I know).

Reply:
The logic to roll-back the database was implemented.